helm-service/controller/deployment_handler.go

```diff
@@ -222,6 +222,16 @@ func (h *DeploymentHandler) getFinishedEventDataForSuccess(inEventData keptnv2.D
 		h.getKeptnHandler().Logger.Info("Using deployment URIs from deployment.triggered event")
 		localURIs = inEventData.Deployment.DeploymentURIsLocal
 		publicURIs = inEventData.Deployment.DeploymentURIsPublic
+	} else if deploymentStrategy == keptnevents.UserManaged {
+		endpoints, err := h.getUserManagedEndpoints(inEventData.EventData)
+		if err != nil {
+			return nil, fmt.Errorf("Could not determine deployment URIs: %s", err.Error())
+		}
+		if endpoints != nil {
+			localURIs = endpoints.DeploymentURIsLocal
+			publicURIs = endpoints.DeploymentURIsPublic
+		}
+
```

Comment:
By having the first else if in line 221, user managed deployments will not take the correct URL once we go over multiple stages with user managed deployments.

For instance, when we deploy in dev and staging and prod, this is the `deployment.triggered` event that shipyard-controller is sending (in Keptn 0.8.2) for stage staging:
```
{
  "data": {
    "configurationChange": {...},
    "deployment": {
      "deploymentNames": [
        "direct"
      ],
      "deploymentURIsLocal": [
        "http://carts.ck-sockshop-dev:80"
      ],
      "deploymentURIsPublic": [
        "https://carts.ck-sockshop-dev.redacted-host.com:443"
      ],
      "deploymentstrategy": "blue_green_service",
      "gitCommit": "3645c6b678c7fd315f53f528bf8fc2c39d89fb48"
    },
    "evaluation": { ... },
    "message": "",
    "project": "ck-sockshop",
    "release": { ... },
    "result": "",
    "service": "carts",
    "stage": "staging",
    "status": "",
    "test": { ... }
  },
  "id": "1a958424-72b0-4a84-b332-3be1d195831f",
  "source": "shipyard-controller",
  "specversion": "1.0",
  "time": "2021-05-12T09:31:39.929Z",
  "type": "sh.keptn.event.deployment.triggered",
  "shkeptncontext": "6067543e-e686-4eb3-aaee-4d829cd941e4",
  "shkeptnspecversion": "0.2.0"
}
```